### DISCUSSION TITLE
Uninstall Hook Not Triggering for Pre-existing Stores After Changing App Hosting and Domain

### DISCUSSION THREAD

**Discussion by smshahriar10** [Kyon147/laravel-shopify#355] [General]
After migrating my Shopify app to a new hosting provider and changing the domain, I’ve encountered an issue. The uninstall webhook is no longer triggering for stores that had installed the app before the hosting and domain change. However the unintall webhook works fine for new installations after the migration. Has anyone experienced a similar issue, or does anyone know why the uninstall hook might not be firing for pre-existing stores? Any advice or suggestions on how to fix this would be greatly appreciated!

**Comment by Kyon147** [comment]
This is because they will have a webhook set up for your old domain.

You would need to create a custom job or something to go through your users and update the webhook manually.

**Comment by chintankotadia** [comment]
I have created a job to resync webhooks and this might be helpful for re-installation of the webhooks.

```
<?php

namespace App\Jobs;

use Illuminate\Bus\Queueable;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Foundation\Bus\Dispatchable;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Queue\SerializesModels;
use Osiset\ShopifyApp\Messaging\Jobs\WebhookInstaller;
use App\Models\User;

class SyncShopWebhookJob implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;

    /**
     * Create a new job instance.
     */
    public function __construct()
    {
        //
    }

    /**
     * Execute the job.
     */
    public function handle(): void
    {
        $user = User::latest('id')->chunk(10, function($users) {
            foreach ($users as $user) {
                dispatch(new WebhookInstaller($user->getId(), config('shopify-app.webhooks')));
            }
        });
    }
}
```

**Comment by smshahriar10** [comment]
Thanks @Kyon147 and @chintankotadia . I solved the problem using your insights.
